sympy/parsing/tests/test_c_parser.py

```diff
@@ -646,6 +647,426 @@ def test_parse():
             )
         )
 
+
+    def test_binary_operators():
+        c_src1 = (
+            'void func()'+
+            '{' + '\n' +
+                'int a;' + '\n' +
+                'a = 1;' + '\n' +
+            '}'
+        )
+        c_src2 = (
+            'void func()'+
+            '{' + '\n' +
+                'int a = 0;' + '\n' +
+                'a = a + 1;' + '\n' +
+                'a = 3*a - 10;' + '\n' +
+            '}'
+        )
+        c_src3 = (
+            'void func()'+
+            '{' + '\n' +
+                'int a = 10;' + '\n' +
+                'a = 1 + a - 3 * 6;' + '\n' +
+            '}'
+        )
+        c_src4 = (
+            'void func()'+
+            '{' + '\n' +
+                'int a;' + '\n' +
+                'int b;' + '\n' +
+                'a = 100;' + '\n' +
+                'b = a*a + a*a + a + 19*a + 1 + 24;' + '\n' +
+            '}'
+        )
+        c_src5 = (
+            'void func()'+
+            '{' + '\n' +
+                'int a;' + '\n' +
+                'int b;' + '\n' +
+                'int c;' + '\n' +
+                'int d;' + '\n' +
+                'a = 1;' + '\n' +
+                'b = 2;' + '\n' +
+                'c = b;' + '\n' +
+                'd = ((a+b)*(a+c))*((c-d)*(a+c));' + '\n' +
+            '}'
+        )
+        c_src6 = (
+            'void func()'+
+            '{' + '\n' +
+                'int a;' + '\n' +
+                'int b;' + '\n' +
+                'int c;' + '\n' +
+                'int d;' + '\n' +
+                'a = 1;' + '\n' +
+                'b = 2;' + '\n' +
+                'c = 3;' + '\n' +
+                'd = (a*a*a*a + 3*b*b + b + b + c*d);' + '\n' +
+            '}'
+        )
+        c_src7 = (
+            'void func()'+
+            '{' + '\n' +
+                'float a;' + '\n' +
+                'a = 1.01;' + '\n' +
+            '}'
+        )
+
+        c_src8 = (
+            'void func()'+
+            '{' + '\n' +
+                'float a;' + '\n' +
+                'a = 10.0 + 2.5;' + '\n' +
+            '}'
+        )
+
+        c_src9 = (
+            'void func()'+
+            '{' + '\n' +
+                'float a = 10.0 + 2.5;' + '\n' +
+                'float b = a * 20.0;' + '\n' +
+            '}'
+        )
+
+        c_src10 = (
+            'void func()'+
+            '{' + '\n' +
+                'int b;' + '\n' +
+                'int c;' + '\n' +
+                'int a = b + c*4;' + '\n' +
```

Comment:
Can you add a couple more examples of these VarDecl statements

Reply:
Also, might be a good idea to add these to `test_var_del`

Reply:
Okay, but there is no function called `test_var_decl`, am I supposed to create a new one?

Reply:
Sure

Reply:
Okay...